releasey/02-create-release-branch.sh

```diff
@@ -0,0 +1,201 @@
+if [[ -z ${version} ]]; then
+  print_error "Missing version"
+  usage >&2
+  exit 1
+fi
+
+# Validate that the commit exists
+if ! git rev-parse --verify "${commit}" >/dev/null 2>&1; then
+  print_error "Invalid Git commit: ${commit}"
+  usage >&2
+  exit 1
+fi
+
+# Validate version format: x.y.z-incubating-rcN
+# TODO: Remove incubating when we are a TLP
+if ! validate_and_extract_rc_version "${version}"; then
+  print_error "Invalid version format. Expected: x.y.z-incubating-rcN where N>0, got: ${version}"
+  usage >&2
+  exit 1
+fi
+
+# Define polaris_version from extracted components
+polaris_version="${major}.${minor}.${patch}-incubating"
+release_branch="release/${polaris_version}"
+
+# Handle RC > 1 scenarios
+if [[ ${rc_number} -gt 1 ]]; then
+  # Check if release branch already exists
+  if git show-ref --verify --quiet "refs/heads/${release_branch}"; then
+    print_info "RC${rc_number} detected and release branch ${release_branch} already exists."
+    print_info "This script only creates release branches for RC1. Nothing to do."
+    exit 0
+  else
+    print_error "RC${rc_number} detected but release branch ${release_branch} does not exist."
+    exit 1
+  fi
+fi
+
+print_info "Starting release branch creation..."
+print_info "Version: ${version}"
+print_info "Polaris version: ${polaris_version}"
+print_info "From commit: ${commit}"
+echo
+
+# Check if release branch already exists
+if git show-ref --verify --quiet "refs/heads/${release_branch}"; then
+  if [[ "${recreate}" == "true" ]]; then
+    print_info "Release branch ${release_branch} already exists, deleting it..."
+    exec_process git branch -D "${release_branch}"
+    if git show-ref --verify --quiet "refs/remotes/${APACHE_REMOTE_NAME}/${release_branch}"; then
+      print_info "Deleting remote release branch from ${APACHE_REMOTE_NAME}..."
+      exec_process git push "${APACHE_REMOTE_NAME}" --delete "${release_branch}"
+    fi
+  else
+    print_error "Release branch ${release_branch} already exists. Use --recreate to delete and recreate it."
+    exit 1
+  fi
+fi
+
+print_info "Checking out commit: ${commit}"
+exec_process git checkout "${commit}"
+
+print_info "Creating release branch: ${release_branch}"
+exec_process git branch "${release_branch}"
+
+print_info "Pushing release branch to ${APACHE_REMOTE_NAME}"
+exec_process git push "${APACHE_REMOTE_NAME}" "${release_branch}" --set-upstream
+
+print_info "Checking out release branch"
+exec_process git checkout "${release_branch}"
+
+print_info "Setting version to ${polaris_version} in version.txt"
+update_version "${polaris_version}"
+
+print_info "Committing and pushing version change"
+exec_process git add "$VERSION_FILE" "$HELM_CHART_YAML_FILE" "$HELM_README_FILE" "$HELM_VALUES_FILE"
+exec_process git commit -m "[chore] Bump version to ${polaris_version} for release"
+exec_process git push
+
+print_info "Updating CHANGELOG.md"
+exec_process cd "${releasey_dir}/.."
+exec_process ./gradlew patchChangelog
```

Comment:
This script exists at line 139 in case of RC2+. Should we re-run the changelog for any RC2+?